src/cov.jl

```diff
@@ -130,15 +130,27 @@ end
 Compute the correlation matrix from the covariance matrix `C` and, optionally, a vector
 of standard deviations `s`. Use `StatsBase.cov2cor!` for an in-place version.
 """
-cov2cor(C::AbstractMatrix, s::AbstractArray=sqrt.(view(C, diagind(C)))) = cov2cor!(copy(C), s)
+cov2cor(C::AbstractMatrix, s::AbstractArray) = cov2cor!(copyto!(similar(C), C), s)
+function cov2cor(C::LinearAlgebra.HermOrSym, s::AbstractArray)
+    D = copyto!(similar(C), C)
+    cov2cor!(parent(D), s)
+    return D
+end
```

Comment:
Wouldn't it make sense to preserve the structure and only work on the active triangle?

Reply:
> My proposal is to return Symmetric and just leave the inactive part alone.

In `StatsBase.cov2cor` by reimplementing the algorithm and not falling back to `Statistics.cov2cor!`? Or in `Statistics.cov2cor!`? I wanted to avoid the latter since I didn't want to wait for a new Julia release.

Reply:
I just saw https://github.com/JuliaStats/StatsBase.jl/issues/444#issuecomment-1227725221. Maybe that's another possible approach - we could copy `cov2cor!` from Statistics to StatsBase and add the optimizations here?

Reply:
I tried to address all comments in this thread in https://github.com/JuliaStats/StatsBase.jl/pull/953/commits/5e01de4e400be18a15820045305fb270b944a04a: The not-in-place methods now uses the correct element type for the output, `cor2cov!` is more efficient, `Symmetric` dispatches are restricted to `Real`s to prevent silently incorrect results, and for `Symmetric` and `Hermitian` only the active triangle is used in the computations.